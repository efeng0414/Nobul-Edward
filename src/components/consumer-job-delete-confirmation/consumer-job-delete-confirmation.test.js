import React from "react";
import renderer from "react-test-renderer";
import ConsumerJobDeleteConfirmation from "./index";

describe("ConsumerJobDeleteConfirmation", () => {
  const componentProps = {
    closeModal: () => {},
    deleteJob: () => {},
    jobType: "buy",
    jobId: "abc123",
    intl: global.getIntl()
  };
  const componentJSX = <ConsumerJobDeleteConfirmation {...componentProps} />;

  describe("Component Snapshot", () => {
    const componentSnapshot = renderer.create(componentJSX).toJSON();
    it("should match existing snapshot", () => {
      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Component render", () => {
    const functionsToReset = ["clickHandle"];
    const componentPrototypeResets = functionsToReset.reduce(
      global.createComponentPrototypeResets({
        component: ConsumerJobDeleteConfirmation
      }),
      {}
    );
    const componentMount = Enzyme.mount(
      componentJSX,
      global.getIntlContext({ intl: componentProps.intl })
    );
    beforeEach(() => {
      functionsToReset.map(
        global.resetPropertyComponentProperty({
          component: ConsumerJobDeleteConfirmation,
          componentMount,
          componentPrototypeResets
        })
      );
    });
    it("should call clickHandle on buttonClick", () => {
      const mockClickHandle = jest.fn(() => Promise.resolve());
      componentMount.instance().clickHandle = mockClickHandle;
      componentMount.setProps({ a: 1 });
      componentMount
        .find(".modal-confirm-button")
        .find("Button")
        .simulate("click");
      expect(mockClickHandle).toHaveBeenCalled();
    });

    it("should call props.deleteJob when clickHandle is called", () => {
      const mockDeletejob = jest.fn(() => Promise.resolve());
      componentMount.setProps({ deleteJob: mockDeletejob });
      componentMount
        .find(".modal-confirm-button")
        .find("Button")
        .simulate("click");
      expect(mockDeletejob).toHaveBeenCalledWith({
        jobId: "abc123",
        jobType: "buy"
      });
    });
    it("should call closeModal function when clickHandle is called", async () => {
      const fakePromise = Promise;
      const mockCloseModal = jest.fn(() => fakePromise.resolve());
      componentMount.setProps({ closeModal: mockCloseModal });
      componentMount
        .find(".modal-confirm-button")
        .find("Button")
        .simulate("click");
      await Promise.all([fakePromise]);
      expect(mockCloseModal).toHaveBeenCalled();
    });
  });
});
