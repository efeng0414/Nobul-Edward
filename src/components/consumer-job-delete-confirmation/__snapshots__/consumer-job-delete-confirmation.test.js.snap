// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerJobDeleteConfirmation Component Snapshot should match existing snapshot 1`] = `
<div
  className="modal-confirm"
>
  <div
    className="modal-confirm-text"
  >
    <p>
      Are you sure you want to delete this posting?
    </p>
  </div>
  <div
    className="modal-confirm-button"
  >
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        Confirm
      </span>
    </button>
  </div>
</div>
`;
